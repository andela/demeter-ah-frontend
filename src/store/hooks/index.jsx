import { useEffect } from 'react';
import { getUser } from '../actions/signup';

export const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return [];
};
