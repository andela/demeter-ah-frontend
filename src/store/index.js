import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import combineReducers from './reducers';
import dummyAction from './actions/Signup';

const middlewares = applyMiddleware(thunk);

const store = createStore(
  combineReducers,
  composeWithDevTools(middlewares),
);

store.dispatch(dummyAction);

export default store;
