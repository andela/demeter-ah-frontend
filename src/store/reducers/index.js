import { combineReducers } from 'redux';
import auth from './Auth';
import articleListing from './articleListing';
import resetPassword from './resetPassword';
import changePassword from './changePassword';
import articles from './articles';
import editProfile from './editProfile';
import viewProfile from './viewProfile';
import membership from './Membership';
import viewArticle from './viewArticle';
import relatedArticles from './relatedArticles';
import bookmarks from './bookmarks';
import bookmarkArticle from './bookmarkArticle';
import comments from './Comments';
import search from './search';
import notifications from './notifications';
import rate from './rateArticle';

const reducers = combineReducers({
  auth,
  resetPassword,
  changePassword,
  articles,
  editProfile,
  viewProfile,
  membership,
  articleListing,
  viewArticle,
  relatedArticles,
  bookmarks,
  bookmarkArticle,
  comments,
  search,
  notifications,
  rate
});

export default reducers;
