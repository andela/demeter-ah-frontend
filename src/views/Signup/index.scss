.signup{
  font-family: 'Montserrat', sans-serif;
  .vline{
    @apply w-px bg-gray-150;
    height: 48%;
  }
  .left-con{
    @apply flex-grow flex flex-row flex-wrap content-around relative px-5 w-112
  }
  .input-wrap {
    height: 300px;
  }
  .right-con{
    @apply flex-grow flex flex-col items-center flex-wrap relative px-5 w-108
  }
  .input{
    @apply w-full px-2
  }
  .inputcon-full{
    @apply w-full px-0 py-1
  }
  .inputcon-half{
    @apply w-1/2 px-0 py-1
  }
}