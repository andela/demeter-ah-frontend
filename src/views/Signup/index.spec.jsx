import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Signup from './index';
import store from '../../store';

describe('NavBar Component', () => {
  it('Should render without errors', () => {
    const component = shallow(
      <Router>
        <Signup store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  it('Should render along with children componenet', () => {
    const component = mount(
      <Router>
        <Signup store={store} />
      </Router>
    );
    expect(component).toMatchSnapshot();
    expect(component.find('button')).toHaveLength(3);
    expect(component.find('input')).toHaveLength(6);
    expect(component.find('img')).toHaveLength(2);
  });
});
