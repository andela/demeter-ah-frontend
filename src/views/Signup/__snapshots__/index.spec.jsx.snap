// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Component Should render along with children componenet 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(Signup)
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Signup
        cleanup={[Function]}
        error={null}
        isAuthenticated={false}
        isCompleted={false}
        isSubmit={false}
        onSignUp={[Function]}
        socialSignUp={[Function]}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        user={Object {}}
      >
        <div
          className="signup w-12/12 md:w-auto"
        >
          <div
            className="bg card-wrapper flex justify-center md:items-center"
          >
            <div
              className="card md:px-8 flex-col w-auto px-2 py-6 m-10 mx-auto max-w-108 sm:max-w-108 md:max-w-220 "
            >
              <h1
                className="w-full mb-4 md:m-0 px-5 text-center md:text-left text-xl md:text-2xl text-purple-650 font-semibold"
              >
                Sign up
              </h1>
              <div
                className="flex flex-col md:flex-row"
              >
                <form
                  className="left-con md:w-112 order-3 md:order-1"
                  onSubmit={[Function]}
                >
                  <div
                    className="md:pr-8 flex flex-col justify-center w-full mt-4"
                  >
                    <InputForm
                      classes="inputcon-full mb-4 pl-0"
                      id="email"
                      inputClass="input text-sm"
                      inputType="text"
                      labelClass="block mb-1 text-sm pl-1"
                      labelname="Email"
                      name="email"
                      onChange={[Function]}
                      pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                      placeholder="johndoe@examle.com"
                      title="must be a valid email"
                    >
                      <div
                        className="inputcon-full mb-4 pl-0"
                      >
                        <Label
                          classes="block mb-1 text-sm pl-1"
                          labelname="Email"
                          name="email"
                        >
                          <label
                            className="block mb-1 text-sm pl-1"
                            htmlFor="email"
                          >
                            Email
                          </label>
                        </Label>
                        <Input
                          classes="input text-sm"
                          id="email"
                          inputType="text"
                          name="email"
                          onChange={[Function]}
                          pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                          placeholder="johndoe@examle.com"
                          title="must be a valid email"
                        >
                          <input
                            className="forminput rounded py-2 bg-purple-150 input text-sm"
                            id="email"
                            name="email"
                            onChange={[Function]}
                            pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                            placeholder="johndoe@examle.com"
                            required={true}
                            title="must be a valid email"
                            type="text"
                          />
                        </Input>
                      </div>
                    </InputForm>
                    <div
                      className="block md:flex justify-center align-center"
                    >
                      <InputForm
                        classes="block inputcon-full mb-4 mr-2"
                        id="firstName"
                        inputClass="input text-sm"
                        inputType="text"
                        labelClass="block mb-1 text-sm pl-1"
                        labelname="First-name"
                        name="firstName"
                        onChange={[Function]}
                        pattern="^[\\\\w]{3,20}$"
                        placeholder="First Name"
                        title="first name is required and must be more than 3 character"
                      >
                        <div
                          className="block inputcon-full mb-4 mr-2"
                        >
                          <Label
                            classes="block mb-1 text-sm pl-1"
                            labelname="First-name"
                            name="firstName"
                          >
                            <label
                              className="block mb-1 text-sm pl-1"
                              htmlFor="firstName"
                            >
                              First-name
                            </label>
                          </Label>
                          <Input
                            classes="input text-sm"
                            id="firstName"
                            inputType="text"
                            name="firstName"
                            onChange={[Function]}
                            pattern="^[\\\\w]{3,20}$"
                            placeholder="First Name"
                            title="first name is required and must be more than 3 character"
                          >
                            <input
                              className="forminput rounded py-2 bg-purple-150 input text-sm"
                              id="firstName"
                              name="firstName"
                              onChange={[Function]}
                              pattern="^[\\\\w]{3,20}$"
                              placeholder="First Name"
                              required={true}
                              title="first name is required and must be more than 3 character"
                              type="text"
                            />
                          </Input>
                        </div>
                      </InputForm>
                      <InputForm
                        classes="block inputcon-full mb-4 md:ml-2"
                        id="lastName"
                        inputClass="input text-sm"
                        inputType="text"
                        labelClass="block mb-1 text-sm pl-1"
                        labelname="Last-name"
                        name="lastName"
                        onChange={[Function]}
                        pattern="^[\\\\w]{3,20}$"
                        placeholder="Last Name"
                        title="last name is required and must be more than 3 character"
                      >
                        <div
                          className="block inputcon-full mb-4 md:ml-2"
                        >
                          <Label
                            classes="block mb-1 text-sm pl-1"
                            labelname="Last-name"
                            name="lastName"
                          >
                            <label
                              className="block mb-1 text-sm pl-1"
                              htmlFor="lastName"
                            >
                              Last-name
                            </label>
                          </Label>
                          <Input
                            classes="input text-sm"
                            id="lastName"
                            inputType="text"
                            name="lastName"
                            onChange={[Function]}
                            pattern="^[\\\\w]{3,20}$"
                            placeholder="Last Name"
                            title="last name is required and must be more than 3 character"
                          >
                            <input
                              className="forminput rounded py-2 bg-purple-150 input text-sm"
                              id="lastName"
                              name="lastName"
                              onChange={[Function]}
                              pattern="^[\\\\w]{3,20}$"
                              placeholder="Last Name"
                              required={true}
                              title="last name is required and must be more than 3 character"
                              type="text"
                            />
                          </Input>
                        </div>
                      </InputForm>
                    </div>
                    <InputForm
                      classes="inputcon-full mb-4"
                      id="username"
                      inputClass="input text-sm"
                      inputType="text"
                      labelClass="block mb-1 text-sm pl-1"
                      labelname="Username"
                      name="username"
                      onChange={[Function]}
                      pattern="^[\\\\w]{3,20}$"
                      placeholder="username"
                      title="username is required and must be more than 3 character"
                    >
                      <div
                        className="inputcon-full mb-4"
                      >
                        <Label
                          classes="block mb-1 text-sm pl-1"
                          labelname="Username"
                          name="username"
                        >
                          <label
                            className="block mb-1 text-sm pl-1"
                            htmlFor="username"
                          >
                            Username
                          </label>
                        </Label>
                        <Input
                          classes="input text-sm"
                          id="username"
                          inputType="text"
                          name="username"
                          onChange={[Function]}
                          pattern="^[\\\\w]{3,20}$"
                          placeholder="username"
                          title="username is required and must be more than 3 character"
                        >
                          <input
                            className="forminput rounded py-2 bg-purple-150 input text-sm"
                            id="username"
                            name="username"
                            onChange={[Function]}
                            pattern="^[\\\\w]{3,20}$"
                            placeholder="username"
                            required={true}
                            title="username is required and must be more than 3 character"
                            type="text"
                          />
                        </Input>
                      </div>
                    </InputForm>
                    <div
                      className="block md:flex justify-center align-center"
                    >
                      <InputForm
                        autoComplete="off"
                        classes="block inputcon-full mb-4 mr-2"
                        id="password"
                        inputClass="input text-sm"
                        inputType="password"
                        labelClass="block mb-1 text-sm pl-1"
                        labelname="Password"
                        name="password"
                        onChange={[Function]}
                        pattern="^[\\\\w]{8,20}$"
                        placeholder="Password"
                        title="password is required and must be more than 8 character"
                      >
                        <div
                          className="block inputcon-full mb-4 mr-2"
                        >
                          <Label
                            classes="block mb-1 text-sm pl-1"
                            labelname="Password"
                            name="password"
                          >
                            <label
                              className="block mb-1 text-sm pl-1"
                              htmlFor="password"
                            >
                              Password
                            </label>
                          </Label>
                          <Input
                            autoComplete="off"
                            classes="input text-sm"
                            id="password"
                            inputType="password"
                            name="password"
                            onChange={[Function]}
                            pattern="^[\\\\w]{8,20}$"
                            placeholder="Password"
                            title="password is required and must be more than 8 character"
                          >
                            <input
                              autoComplete="off"
                              className="forminput rounded py-2 bg-purple-150 input text-sm"
                              id="password"
                              name="password"
                              onChange={[Function]}
                              pattern="^[\\\\w]{8,20}$"
                              placeholder="Password"
                              required={true}
                              title="password is required and must be more than 8 character"
                              type="password"
                            />
                          </Input>
                        </div>
                      </InputForm>
                      <InputForm
                        autoComplete="off"
                        classes="block inputcon-full mb-4 md:ml-2"
                        id="confirmPassword"
                        inputClass="input text-sm"
                        inputType="password"
                        labelClass="block mb-1 text-sm pl-1"
                        labelname="Confirm Password"
                        name="confirmPassword"
                        onChange={[Function]}
                        placeholder="Confirm Password"
                      >
                        <div
                          className="block inputcon-full mb-4 md:ml-2"
                        >
                          <Label
                            classes="block mb-1 text-sm pl-1"
                            labelname="Confirm Password"
                            name="confirmPassword"
                          >
                            <label
                              className="block mb-1 text-sm pl-1"
                              htmlFor="confirmPassword"
                            >
                              Confirm Password
                            </label>
                          </Label>
                          <Input
                            autoComplete="off"
                            classes="input text-sm"
                            id="confirmPassword"
                            inputType="password"
                            name="confirmPassword"
                            onChange={[Function]}
                            placeholder="Confirm Password"
                          >
                            <input
                              autoComplete="off"
                              className="forminput rounded py-2 bg-purple-150 input text-sm"
                              id="confirmPassword"
                              name="confirmPassword"
                              onChange={[Function]}
                              placeholder="Confirm Password"
                              required={true}
                              type="password"
                            />
                          </Input>
                        </div>
                      </InputForm>
                    </div>
                  </div>
                  <div
                    className="w-full flex flex-col items-center md:flex-row justify-end"
                  >
                    <p
                      className="flex-grow order-last md:order-first flex items-center my-0 md:my-8 text-sm"
                    >
                      Have an account? 
                      <Link
                        className="text-purple-650 ml-2 font-semibold cursor-pointer"
                        to="/signin"
                      >
                        <a
                          className="text-purple-650 ml-2 font-semibold cursor-pointer"
                          href="/signin"
                          onClick={[Function]}
                        >
                          Sign in
                        </a>
                      </Link>
                    </p>
                    <Button
                      classes="btn-purple w-1/3 md:mx-8 my-4 md:my-7"
                      id="register"
                      isSubmit={false}
                      name="Register"
                      type="submit"
                    >
                      <button
                        className="btn btn-purple w-1/3 md:mx-8 my-4 md:my-7"
                        disabled={false}
                        id="register"
                        type="submit"
                      >
                        Register
                      </button>
                    </Button>
                  </div>
                </form>
                <div
                  className="order-2 flex flex-grow flex-col items-center relative"
                >
                  <hr
                    className="vline hidden md:block"
                  />
                  <span>
                    OR
                  </span>
                  <hr
                    className="vline hidden md:block"
                  />
                </div>
                <div
                  className="order-1 md:order-3 right-con md:w-108 justify-center"
                >
                  <img
                    alt="Authors Haven"
                    className="hidden md:block"
                    id="brand-logo"
                    src="/brand-logo.png"
                  />
                  <Button
                    classes="w-64 px-4 py-2 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                    id="google"
                    name="Login"
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="btn w-64 px-4 py-2 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                      disabled={false}
                      id="google"
                      onClick={[Function]}
                      type="button"
                    >
                      <img
                        alt="Authors Haven"
                        src="/google.png"
                      />
                      Sign up with Google
                    </button>
                  </Button>
                  <Button
                    classes="w-64 px-4 py-2 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                    id="facebook"
                    name="Login"
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="btn w-64 px-4 py-2 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                      disabled={false}
                      id="facebook"
                      onClick={[Function]}
                      type="button"
                    >
                      <img
                        alt="Authors Haven"
                        src="/facebook.png"
                      />
                      Sign up with Facebook
                    </button>
                  </Button>
                  <Button
                    classes="w-64 px-4 py-2 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                    id="twitter"
                    name="Login"
                    onClick={[Function]}
                    type="button"
                  >
                    <button
                      className="btn w-64 px-4 py-2 flex items-center justify-around shadow-md rounded-lg mx-2 my-3 text-sm"
                      disabled={false}
                      id="twitter"
                      onClick={[Function]}
                      type="button"
                    >
                      <img
                        alt="Authors Haven"
                        src="/twitter.png"
                      />
                      Sign up with Twitter
                    </button>
                  </Button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </Signup>
    </Connect(Signup)>
  </Router>
</BrowserRouter>
`;

exports[`Sign up Component Should render without errors 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(Signup)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</Router>
`;
